lib/ffmpegCommandService.js

```diff
@@ -406,6 +397,63 @@ class FfmpegCommandService {
         return {filter, maps};
     }
 
+    _createFilterForTextOverlay(filter, textOverlay, i) {
+        const bashCharsToEscapeRegex = /!/g;
+
+        // @TODO: add handling for escaping special characters in bash.  Know it's at least ! that needs escaping.
+        
+        const textAlignment = textOverlay.textAlignment || null;
+
+        const x = this._calculateHorizontalPosition(textAlignment ? textAlignment.horizontal : null, textOverlay.xLoc);
+        const y = this._calculateVerticalPosition(textAlignment ? textAlignment.vertical : null, textOverlay.yLoc);
+
+        const fontFile = Path.resolve(this._fontConfig.fontsDirectory, this._fontLookupObj[textOverlay.fontName].fontFilePath);
+        filter += `${i > 0 ? ',' : ''} drawtext=enable=1:text='${textOverlay.text.replace(bashCharsToEscapeRegex, "\\$&")}':x=${x}:y=${y}`;
```

Comment:
I would factor out `textOverlay.text.replace(bashCharsToEscapeRegex, "\\$&")` into a variable to make this more readable.